testing/src/scenario/_consistency_checker.py

```diff
@@ -170,16 +170,14 @@ def check_event_consistency(
     errors: List[str] = []
     warnings: List[str] = []
 
+    # custom event: can't make assumptions about its name and its semantics
+    # todo: should we then just skip the other checks?
     if not event._is_builtin_event(charm_spec):
-        # This is a custom event - we can't make assumptions about its name and
-        # semantics. It doesn't really make sense to do checks that are designed
-        # for relations, workloads, and so on - most likely those will end up
-        # with false positives. Realistically, we can't know about what the
-        # requirements for the custom event are (in terms of the state), so we
-        # skip everything here. Perhaps in the future, custom events could
-        # optionally include some sort of state metadata that made testing
-        # consistency possible?
-        return Results(errors, warnings)
+        warnings.append(
+            "this is a custom event; if its name makes it look like a builtin one "
+            "(e.g. a relation event, or a workload event), you might get some false-negative "
+            "consistency checks.",
+        )
```

Comment:
I like emitting a warning here, but given the comments (old and new), it kind of seems like returning early and skipping the other checks is correct.

Is the idea that we shouldn't be shielding users from the consequences of using a poorly named custom event that could clash with a relation/workload/action/storage event?

Reply:
Yes, there was meant to be the warning and then a return.